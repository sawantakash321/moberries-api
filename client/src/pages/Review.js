import Checkbox from '@material-ui/core/Checkbox';
import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormGroup from '@material-ui/core/FormGroup';
import FormHelperText from '@material-ui/core/FormHelperText';
import Grid from '@material-ui/core/Grid';
import {withStyles} from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import React from 'react';

const payment = {
  true: 'Yes',
  false: 'No',
};

const styles = theme => ({
  listItem: {
    padding: `${theme.spacing.unit}px 0`,
  },
  total: {
    fontWeight: '700',
  },
  title: {
    marginTop: theme.spacing.unit * 2,
  },
});

const Review = props => {
  const {
    classes,
    values,
    handleChange,
    errors: {errors},
  } = props;
  return (
    <React.Fragment>
      <Typography variant="h4" gutterBottom>
        Order summary
      </Typography>
      <Typography variant="h6" gutterBottom>
        Duration: {`${values.duration} Months`}
      </Typography>
      <Typography variant="h6" gutterBottom>
        Gigabytes: {`${values.gigabytes} GB`}
      </Typography>
      <Typography variant="h6" gutterBottom>
        Upfront Payment: {payment[values.upfrontPayment]}
      </Typography>
      <Typography variant="h6" gutterBottom>
        Total Price: {`$${values.computedPrice}`}
      </Typography>
      <br />
      <br />

      <Grid container spacing={16}>
        <Grid item xs={12} sm={6}>
          <Typography variant="h4" gutterBottom className={classes.title}>
            User Details
          </Typography>
          <Typography variant="h6" gutterBottom>{`${values.firstName} ${values.lastName}`}</Typography>
          <Typography variant="h6" gutterBottom>
            {values.email}
          </Typography>
          <Typography variant="h6" gutterBottom>
            {values.streetAddr}
          </Typography>
        </Grid>

        <Grid item container direction="column" xs={12} sm={6}>
          <Typography variant="h4" gutterBottom className={classes.title}>
            Payment details
          </Typography>
          <Grid container>
            <React.Fragment key={values.creditNumber}>
              <Grid item xs={6}>
                <Typography variant="h6" gutterBottom>
                  Card number
                </Typography>
              </Grid>
              <Grid item xs={6}>
                <Typography variant="h6" gutterBottom>
                  {values.creditNumber}
                </Typography>
              </Grid>
              <Grid item xs={6}>
                <Typography variant="h6" gutterBottom>
                  Expiry date
                </Typography>
              </Grid>
              <Grid item xs={6}>
                <Typography variant="h6" gutterBottom>
                  {values.expDate}
                </Typography>
              </Grid>
              <Grid item xs={6}>
                <Typography variant="h6" gutterBottom>
                  CVV
                </Typography>
              </Grid>
              <Grid item xs={6}>
                <Typography variant="h6" gutterBottom>
                  {values.cvv}
                </Typography>
              </Grid>
            </React.Fragment>
          </Grid>
        </Grid>
        <br />
        <br />
        <Grid item xs={12}>
          <FormControl required error={!errors.isValid} component="fieldset" className={classes.formControl}>
            <FormGroup>
              <FormControlLabel
                control={
                  <Checkbox
                    checked={values.checked}
                    color="secondary"
                    name="termsCheckbox"
                    onChange={handleChange('checked')}
                  />
                }
                label="I have read and accepted the Terms Agreement"
              />
            </FormGroup>
            <FormHelperText>{errors.errors.checked}</FormHelperText>
          </FormControl>
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

Review.propTypes = {
  classes: PropTypes.object.isRequired,
  values: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default withStyles(styles)(Review);
