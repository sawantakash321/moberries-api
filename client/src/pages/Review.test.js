'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import Review from './Review';

// Snapshot test
it('matches the snapshot when there are no validation errors', () => {
  const values = {
    firstName: 'Akash',
    lastName: 'Sawant',
    email: 'sawantakash321@gmail.com',
    streetAddr: 'MLK street, behind cathom mart, Mumbai',
    duration: '6',
    gigabytes: '20',
    upfrontPayment: 'true',
    computedPrice: 6.33,
    creditNumber: '4565456543764357',
    expDate: '012022',
    cvv: '765',
    checked: true,
  };
  const errors = {
    errors: {
      checked: '',
    },
    isValid: true,
  };
  const handleChange = jest.fn();
  const tree = renderer.create(<Review values={values} errors={{errors}} handleChange={handleChange} />).toJSON();
  expect(tree).toMatchSnapshot();
});
