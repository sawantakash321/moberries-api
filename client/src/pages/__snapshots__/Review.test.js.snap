// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot when there are no validation errors 1`] = `
Array [
  <h4
    className="MuiTypography-root-4 MuiTypography-h4-19 MuiTypography-gutterBottom-31"
  >
    Order summary
  </h4>,
  <h6
    className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
  >
    Duration: 
    6 Months
  </h6>,
  <h6
    className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
  >
    Gigabytes: 
    20 GB
  </h6>,
  <h6
    className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
  >
    Upfront Payment: 
    Yes
  </h6>,
  <h6
    className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
  >
    Total Price: 
    $6.33
  </h6>,
  <br />,
  <br />,
  <div
    className="MuiGrid-container-40 MuiGrid-spacing-xs-16-63"
  >
    <div
      className="MuiGrid-item-41 MuiGrid-grid-xs-12-80 MuiGrid-grid-sm-6-88"
    >
      <h4
        className="MuiTypography-root-4 MuiTypography-h4-19 MuiTypography-gutterBottom-31 Review-title-3"
      >
        User Details
      </h4>
      <h6
        className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
      >
        Akash Sawant
      </h6>
      <h6
        className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
      >
        sawantakash321@gmail.com
      </h6>
      <h6
        className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
      >
        MLK street, behind cathom mart, Mumbai
      </h6>
    </div>
    <div
      className="MuiGrid-container-40 MuiGrid-item-41 MuiGrid-direction-xs-column-43 MuiGrid-grid-xs-12-80 MuiGrid-grid-sm-6-88"
    >
      <h4
        className="MuiTypography-root-4 MuiTypography-h4-19 MuiTypography-gutterBottom-31 Review-title-3"
      >
        Payment details
      </h4>
      <div
        className="MuiGrid-container-40"
      >
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-6-74"
        >
          <h6
            className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
          >
            Card number
          </h6>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-6-74"
        >
          <h6
            className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
          >
            4565456543764357
          </h6>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-6-74"
        >
          <h6
            className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
          >
            Expiry date
          </h6>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-6-74"
        >
          <h6
            className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
          >
            012022
          </h6>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-6-74"
        >
          <h6
            className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
          >
            CVV
          </h6>
        </div>
        <div
          className="MuiGrid-item-41 MuiGrid-grid-xs-6-74"
        >
          <h6
            className="MuiTypography-root-4 MuiTypography-h6-21 MuiTypography-gutterBottom-31"
          >
            765
          </h6>
        </div>
      </div>
    </div>
    <br />
    <br />
    <div
      className="MuiGrid-item-41 MuiGrid-grid-xs-12-80"
    >
      <fieldset
        className="MuiFormControl-root-137"
      >
        <div
          className="MuiFormGroup-root-141"
        >
          <label
            className="MuiFormControlLabel-root-143"
          >
            <span
              className="MuiButtonBase-root-165 MuiIconButton-root-159 MuiPrivateSwitchBase-root-155 MuiCheckbox-root-149 MuiCheckbox-colorSecondary-154 MuiPrivateSwitchBase-checked-156 MuiCheckbox-checked-150"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-164"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-168"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
                <input
                  checked={true}
                  className="MuiPrivateSwitchBase-input-158"
                  data-indeterminate={false}
                  disabled={false}
                  name="termsCheckbox"
                  onChange={[Function]}
                  type="checkbox"
                />
              </span>
              <span
                className="MuiTouchRipple-root-185"
              />
            </span>
            <span
              className="MuiTypography-root-4 MuiTypography-body1-13 MuiFormControlLabel-label-148"
            >
              I have read and accepted the Terms Agreement
            </span>
          </label>
        </div>
        <p
          className="MuiFormHelperText-root-177 MuiFormHelperText-required-184"
        >
          
        </p>
      </fieldset>
    </div>
  </div>,
]
`;
